### ISSUE TITLE
How to calculate total optimization steps

### ISSUE THREAD

**Issue by HelloWorld506** [huggingface/open-r1#483]
I ran it on 8 GPUs and set num_generations to 8, num_processes=7， Why Total optimization steps=196, isn't it Num examples/Total train batch size? It seems that multiplying by num_generations yields 196. Why do we need to multiply by num_generations？
[INFO|trainer.py:2405] 2025-03-06 12:04:09,913 >> ***** Running training *****
[INFO|trainer.py:2406] 2025-03-06 12:04:09,913 >>   Num examples = 5,498
[INFO|trainer.py:2407] 2025-03-06 12:04:09,914 >>   Num Epochs = 1
[INFO|trainer.py:2408] 2025-03-06 12:04:09,914 >>   Instantaneous batch size per device = 8
[INFO|trainer.py:2411] 2025-03-06 12:04:09,914 >>   Total train batch size (w. parallel, distributed & accumulation) = 224
[INFO|trainer.py:2412] 2025-03-06 12:04:09,914 >>   Gradient Accumulation steps = 4
[INFO|trainer.py:2413] 2025-03-06 12:04:09,914 >>   Total optimization steps = 196
[INFO|trainer.py:2414] 2025-03-06 12:04:09,915 >>   Number of trainable parameters = 7,615,616,512

**Comment by 999wwx** [comment]
+1
